packages/ui/src/TextArea.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { ChangeEvent, FC, FocusEvent, ReactElement, useRef } from 'react'
+import { v4 as uuid } from 'uuid'
+import cn from 'classnames'
+
+import { DataTestProp } from '@hazelcast/helpers'
+import { HiddenLabel } from '../src/HiddenLabel'
+import { Error, errorId } from '../src'
+
+import styles from './TextArea.module.scss'
+
+export type TextAreaCoreProps = {
+  name: string
+  value?: string
+  onBlur?: (e: FocusEvent<HTMLTextAreaElement>) => void
+  onChange: (e: ChangeEvent<HTMLTextAreaElement>) => void
+  error?: string
+}
+
+export type TextAreaExtraProps = {
+  label: string
+  inputClassName?: string
+  errorClassName?: string
+  inputContainerChild?: ReactElement
+} & Partial<Pick<HTMLTextAreaElement, 'className' | 'defaultValue' | 'disabled' | 'placeholder' | 'required'>>
+
+export type TextAreaProps = TextAreaCoreProps & TextAreaExtraProps & DataTestProp
+
+export const TextArea: FC<TextAreaProps> = ({
+  className,
+  error,
+  inputClassName,
+  label,
+  name,
+  onBlur,
+  onChange,
+  placeholder,
+  value,
+  defaultValue,
```

Comment:
Do you think we will need to use the textarea as an uncontrolled component?

Reply:
Good point, didn't realise that, thanks!